### PR TITLE
coin-cgl: add coin-cgl/0.60.3 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **coin-cgl/0.60.3**

- [x] Depends on #2872
- [x] Depends on #2873
- [x] Depends on #2874

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — DISMISSED** [review]

**Review by danimtb — APPROVED** [review]

**Comment by madebr** [comment]
@SpaceIm @prince-chrismc 
Can (one of) you please review this?

File: recipes/coin-cgl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+from contextlib import contextmanager
+import os
+
+
+class CoinCglConan(ConanFile):
+    name = "coin-cgl"
+    description = "COIN-OR Cut Generator Library"
+    topics = ("conan", "clp", "simplex", "solver", "linear", "programming")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/coin-or/Clp"
```
Comment (prince-chrismc) [inline recipes/coin-cgl/all/conanfile.py:12]:
```suggestion
    homepage = "https://github.com/coin-or/Cgl"
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/coin-cgl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("coin-utils/2.11.4")
+        self.requires("coin-osi/0.108.6")
+        self.requires("coin-clp/1.17.6")
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/1.7.3")
```
Comment (SpaceIm) [inline recipes/coin-cgl/all/conanfile.py:52]: pkgconf comes for free in msys2 package (as well as autoconf, automake etc), but I know that you prefer to have full control over which pkgconf binary is used inside the recipe.

File: recipes/coin-cgl/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+
+        # FIXME: missing CglLandP.hpp etc headers in include
```
Comment (SpaceIm) [inline recipes/coin-cgl/all/conanfile.py:101]: is it not fixed now?

```
 /usr/bin/install -c -m 644 '/Users/jenkins/w/cci_PR-2875@2/.conan/data/coin-cgl/0.60.3/_/_/build/12380cd09889cf0f93b5621cb258b79493b4c9a5/source_subfolder/Cgl/src/CglLandP/CglLandP.hpp' '/Users/jenkins/w/cci_PR-2875@2/.conan/data/coin-cgl/0.60.3/_/_/package/12380cd09889cf0f93b5621cb258b79493b4c9a5/include/coin/CglLandP.hpp'
```

File: recipes/coin-cgl/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/coin-cgl/all/conanfile.py:52]: As long as `base-devel` is installed: https://packages.msys2.org/package/pkg-config
Which the default msys2 does:
https://github.com/conan-io/conan-center-index/blob/88a19d55ad885c9b7f14918bd428cd424f9d18f3/recipes/msys2/all/conanfile.py#L20

Better be safe then sorrow, like meson and ninja

File: recipes/coin-cgl/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
```
Comment (madebr) [inline recipes/coin-cgl/all/conanfile.py:104]: I guess so. It's a leftover from the pain of packaging this :rofl: 
```suggestion
    def package(self):
        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/coin-cgl/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+
+        os.unlink(os.path.join(self.package_folder, "lib", "libCgl.la"))
+
+        if self.settings.compiler == "Visual Studio":
+            os.rename(os.path.join(self.package_folder, "lib", "libCgl.lib"),
+                      os.path.join(self.package_folder, "lib", "Cgl.lib"))
```
Comment (jgsogo) [inline recipes/coin-cgl/all/conanfile.py:109]: Is this something we want to know or is it related to a bug in Conan that is not managing well the `lib` prefix?

Comment (madebr) [reply recipes/coin-cgl/all/conanfile.py:109]: This is project specific.
Autotools projects always create `libXXX.lib` libraries on MSVC.
Consumers of coin are expected to use the .pc files, so the exact name of the library does not matter.

**Review by jgsogo — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Pull request merged**